src/clients/discover_registration_services.rb

```diff
@@ -28,91 +25,13 @@ def main
 
   private
 
-    def button_box
-      ButtonBox(
-        PushButton(Id(:ok), Opt(:default), Label.OKButton),
-        PushButton(Id(:cancel), Label.CancelButton)
-      )
-    end
-
-    def dialog_content
-      MarginBox(2, 0.5,
-        VBox(
-          # popup heading (in bold)
-          Heading(_("Local Registration Servers")),
-          VSpacing(0.5),
-          Label(_("Select a detected registration server from the list\n" \
-                "or the default SUSE registration server.")),
-          VSpacing(0.5),
-          RadioButtonGroup(
-            Id(:services),
-            Left(
-              HVSquash(
-                VBox(*services_radio_buttons)
-              )
-            )
-          ),
-          VSpacing(Opt(:vstretch), 1),
-          button_box
-        ))
-    end
-
-    def handle_dialog
-      loop do
-        case UI.UserInput
-        when :ok
-          selected = UI.QueryWidget(Id(:services), :CurrentButton)
-          if !selected
-            # error popup
-            Report.Error(_("No registration server selected."))
-            next
-          end
-
-          break if selected == "scc"
-
-          return service_url(selected)
-        when :cancel
-          return :cancel
-        end
-      end
-    end
-
-    def service_url(selected)
-      selected_service = services[selected.to_i]
-      log.info "Selected service #{selected_service.inspect}"
-
-      url = ::Registration::UrlHelpers.service_url(selected_service.slp_url)
+    def select_registration_service
+      service = ::Registration::UI::ServiceSelectionDialog.run(services)
+      return nil unless service.respond_to?(:slp_url)
```

Comment:
ok, seeing logic below I expect here to be full handling...e.g. what happen if user select slp and then go back and select scc? it should be set back, not?